plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/NameEmbeddings.kt

```diff
@@ -31,19 +33,51 @@ data class ClassName(val packageName: FqName, val className: Name) : MangledName
         get() = "pkg_${packageName.asString()}\$class_${className.asString()}"
 }
 
-data class ClassMemberName(val className: ClassName, val name: Name) : MangledName {
+open class ClassMemberName(val className: ClassName, val name: Name) : MangledName {
     override val mangled: String
         get() = "${className.mangled}\$member_${name.asString()}"
 }
 
-/** Global name not associated with a class.
+class ClassMemberAccessor(className: ClassName, name: Name, val accessor: AccessorType) : ClassMemberName(className, name) {
+    enum class AccessorType {
```

Comment:
Instead of going the way and making an enum for this why not just make two data classes `ClassMemberSetter` and `ClassMemberGetter`? Would this simplify the code a bit?